python3/tests/import_helper.py

```diff
@@ -0,0 +1,70 @@
+"""helpers for unit-testing functions in scripts without permanent global mocks"""
+import os
+import sys
+from contextlib import contextmanager
+from types import ModuleType
+
+from typing import Generator
+from mock import Mock
+
+
+@contextmanager
+def mocked_modules(*module_names):  # type:(str) -> Generator[None, None, None]
+    """Context manager that temporarily mocks the specified modules.
+
+    :param module_names: Variable number of names of the modules to be mocked.
+    :yields: None
+
+    During the context, the specified modules are added to the sys.modules
+    dictionary as instances of the ModuleType class.
+    This effectively mocks the modules, allowing them to be imported and used
+    within the context. After the context, the mocked modules are removed
+    from the sys.modules dictionary.
+
+    Example usage:
+    ```python
+    with mocked_modules("module1", "module2"):
+        # Code that uses the mocked modules
+    ```
+    """
+    for module_name in module_names:
+        sys.modules[module_name] = Mock()
+    yield
+    for module_name in module_names:
+        sys.modules.pop(module_name)
+
+
+def import_file_as_module(relative_script_path):  # type:(str) -> ModuleType
```

Comment:
this file is duplicated with https://github.com/xapi-project/xen-api/blob/feature/py3/python3/unittest/import_file.py

Reply:
Not quite, this one provides more functionality

Reply:
I've prepared a branch that deletes import_file.py: https://github.com/xapi-project/xen-api/compare/feature/py3...psafont:xen-api:test-and-fix-usb_reset-mount